UglyTicTacToe/Default.aspx.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+using UglyTicTacToe.Models;
+using System.Windows;
+namespace UglyTicTacToe
+{
+    public partial class Default : System.Web.UI.Page
+    {
+        MyState lastState = Buttons.recentState;
+        MyState currentState;
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!IsPostBack)
+            {
+                GameLogic.initArray();
+            }
+        }
+
+        protected void NewGameButton_Click(object sender, EventArgs e)
+        {
+            Buttons.recentState = MyState.None;
+            GameLogic.ButtonsLeft = 9;
+            GameLogic.initArray();
+            Response.Redirect("~/Default.aspx");
+        }
+
+        protected void Button_Click(object sender, EventArgs e)
+        {
+            Button clickedButton = (Button)sender;
+            string buttonIDString = "";
+            clickedButton.Enabled = false;
+            GameLogic.ButtonsLeft -= 1;
+            buttonIDString = clickedButton.ID;
+            if (lastState == MyState.None || lastState == MyState.Nill)
+            {
+                clickedButton.Text = "X";
+                currentState = MyState.Cross;
+                Buttons.recentState = currentState;
+            }
+            else if (lastState == MyState.Cross)
+            {
+                clickedButton.Text = "O";
+                currentState = MyState.Nill;
+                Buttons.recentState = currentState;
+            }
+
+            UpdateGameLogic(buttonIDString);
```

Comment:
Here's where you could pass `currentState` into the `UpdateGameLogic` method and eliminate the `currentState` private field.